File: SchoolWebApp-API/Project.Core/Entities/Academics/Exam.cs
using SchoolWebApp.Core.Entities.Class;
using SchoolWebApp.Core.Entities.Shared;
using System.ComponentModel.DataAnnotations;

namespace SchoolWebApp.Core.Entities.Academics
{
    public class Exam : Base
    {
        public required string Name { get; set; }
        [Required(ErrorMessage = "Enter the exam total mark")]
        [Display(Name = "Examination total mark")]
        public float ExamMark { get; set; }

        [Required(ErrorMessage = "Enter the examination contributing mark")]
        [Display(Name = "Contributing mark")]
        public float ContributingMark { get; set; }

        public string? OtherDetails { get; set; }

        public int ExamTypeId { get; set; }
        public ExamType? ExamType { get; set; }
        public int SchoolClassId { get; set; }
        public SchoolClass? SchoolClass { get; set; }
        public int SessionId { get; set; }
        public Session? Session { get; set; }
        public int SubjectId { get; set; }
        public Subject? Subject { get; set; }

        public List<ExamResult> ExamResults { get; set; } = new();
    }
}

File: SchoolWebApp-Client/src/app/academics/models/exam.ts
import {ResourceModel} from '@/core/models/ResourceModel';
import {ExamType} from './exam-type';
import {SchoolClass} from '@/class/models/school-class';
import {Session} from '@/class/models/session';
import {Subject} from './subject';

export class Exam extends ResourceModel<Exam> {
    public name?: string;
    public examMark?: number;
    public contributingMark?: number;
    public otherDetails?: string;

    public examNameId?: number;
    public examName?: any;

    public examTypeId?: number;
    public examType?: ExamType;

    public schoolClassId?: number;
    public schoolClass?: SchoolClass;

    public sessionId?: number;
    public session?: Session;

    public subjectId?: number;
    public subject?: Subject;

    public examStartDate?: string;
    public examEndDate?: string;
    public examMarkEntryEndDate?: string;

    public description?: string;

    constructor(model?: Partial<Exam>) {
        super(model);
    }
}
